substrate/client/transaction-pool/tests/zombienet/limits_30.rs

```diff
@@ -0,0 +1,164 @@
+				})
+			})
+			.map(|_| true)
+			.unwrap_or(false)
+	}
+
+	fn config(&self) -> Result<NetworkConfig, anyhow::Error> {
+		let config = NetworkConfigBuilder::new()
+			.with_relaychain(|r| {
+				let mut rc_nodes_iter = self.rc_nodes.iter();
+				let first_node = rc_nodes_iter.next();
+				// All nodes should use `polkadot` bin.
+				let rc = r
+					.with_chain(self.rc_config.chain.as_str())
+					.with_default_command(self.rc_config.default_command.as_str())
+					.with_node(|node| {
+						node.with_name(
+							first_node.map(|node| node.name.as_str()).unwrap_or("rc-unamed-0"),
+						)
+						.with_rpc_port(DEFAULT_RC_NODE_RPC_PORT)
+						.validator(first_node.map(|node| node.validator).unwrap_or(false))
+					});
+				(DEFAULT_RC_NODE_RPC_PORT as usize + 1..
+					DEFAULT_RC_NODE_RPC_PORT as usize + self.rc_nodes.len())
+					.zip(rc_nodes_iter)
+					.fold(rc, move |acc, (port, node)| {
+						acc.with_node(|new_node| {
+							new_node
+								.with_name(&node.name)
+								.with_rpc_port(u16::try_from(port).unwrap_or(0))
+								.validator(node.validator)
+						})
+					})
+			})
+			.with_parachain(|p| {
+				let mut pc_nodes_iter = self.pc_nodes.iter();
+				let first_node = pc_nodes_iter.next();
+
+				// Set up the parachain and the first collator, obtaining the right type to iterate
+				// through the following collators through folding.
+				let p = p
+					.with_id(self.pc_config.id)
+					.cumulus_based(true)
+					.with_chain_spec_path(self.pc_config.chain_spec_path.as_str())
+					.with_default_command(self.pc_config.default_command.as_str())
+					.with_collator(|new_node| {
+						new_node
+							.with_name(
+								first_node.map(|node| node.name.as_str()).unwrap_or("pc-unamed-0"),
+							)
+							.with_rpc_port(DEFAULT_PC_NODE_RPC_PORT)
+							.validator(first_node.map(|node| node.validator).unwrap_or(false))
+							.with_args(first_node.map(|node| node.args.clone()).unwrap_or(vec![]))
+					});
+
+				(DEFAULT_PC_NODE_RPC_PORT as usize + 1..
+					DEFAULT_PC_NODE_RPC_PORT as usize + self.pc_nodes.len())
+					.zip(pc_nodes_iter)
+					.fold(p, move |acc, (port, pc_node_config)| {
+						acc.with_collator(|node| {
+							node.with_name(&pc_node_config.name)
+								.with_rpc_port(u16::try_from(port).unwrap_or(0))
+								.validator(pc_node_config.validator)
+								.with_command(self.pc_config.default_command.as_str())
+								.with_args(pc_node_config.args.clone())
+						})
+					})
+			});
+
+		config.build().map_err(|e| {
+			let errs = e.into_iter().map(|e| e.to_string()).collect::<Vec<_>>().join(" ");
+			anyhow!("config errs: {errs}")
+		})
+	}
+
+	async fn start(&self) -> Result<ZNetwork<LocalFileSystem>, anyhow::Error> {
+		let network_config = self.config()?;
+		if !self.ensure_bins_on_path() {
+			return Err(anyhow!("Error: required bins weren't found on $PATH: polkadot"));
+		}
+		network_config.spawn_native().await.map_err(|err| anyhow!(format!("{}", err)))
```

Comment:
Here I have one request:
Would be good to have some way to print the zombienet network params (just like it is done today with toml files). I would also like to see full paths to the binaries.

Enabling logs will be needed in `stand_alone` group of test (which just spawns networks and do no more, see my [comment](https://github.com/paritytech/polkadot-sdk/pull/7257#pullrequestreview-2563818561) for details). For `integrations` test this may not be needed.

Reply:
Printing the binaries path would be very helpful indeed. For cargo tests, we can reference the executable paths somewhere in the base dir of the network (if they aren't already mentioned there) - @pepoviola do you know?

But knowing the network basedir is another aspect with cargo test. We might be able to identify the basedir under `tmp` if we name it accordingly to the test in question, and add a timestamp suffix in the end for uniqueness. It would not be entirely perfect, but I find it usable. WDYT?